zabbix_metrics_watcher.py

```diff
@@ -214,14 +217,32 @@ def run(zabbix_host, osbs_master, config, instance):
 
             logger.info("Found build %s in %s", build_name, status)
             build = Build(build_name, cmd_base)
-            if status == 'Pending':
+            if status == 'New':
+                now = datetime.datetime.now()
+                if build_name not in builds_in_new:
+                    builds_in_new[build_name] = now
+                spent_in_new = [(now - start).total_seconds() for start in builds_in_new.values()]
+                max_spent_in_new = max(spent_in_new)
+                if spent_in_new and max_spent_in_new > STUCK_IN_NEW_THRESHOLD_SECS:
+                    logger.info("%s build(s) are in New for, longest: %s sec",
+                                len(builds_in_new), max_spent_in_new)
+                    _send_zabbix_message(zabbix_host, osbs_master, "new_duration", max_spent_in_new)
+
+            elif status == 'Pending':
+                builds_in_new = builds_in_new.pop(build_name, {})
+                if not builds_in_new:
+                    _send_zabbix_message(zabbix_host, osbs_master, "new_duration", 0)
```

Comment:
>Also the .pop looks suspicious

Its just the same as `del builds_in_new[build_name]`, but it won't throw exceptions and return `{}` instead

>we probably want to being doing post-New procesing for != 'New', not just 'Pending'

Yeah, that's a good idea. I don't think it would skip the Pending state usually (as it has to pull the buildroot), but its possible

>I'm not sure about the STUCK_IN_NEW_THRESHOLD_SECS thing

This protects us from builds which quickly change New state to Pending. Less that 5 secs is considered to be fine